src/libmongoc/src/mongoc/mongoc-topology-scanner-private.h

```diff
@@ -109,6 +110,9 @@ typedef struct mongoc_topology_scanner {
    bool negotiate_sasl_supported_mechs;
    bool bypass_cooldown;
    bool speculative_authentication;
+
+   bson_mutex_t ip_mutex;
+   char *bind_ip;
```

Comment:
The topology scanner is the owner of the custom `bind_ip.`  Pools and clients and the SDAM framework all have access to the topology scanner and so can share the same custom `bind_ip` if it is stored here.